Adds support for Linux and windows

MoAlyousef opened the pull request:
Hello @AllenDang 

This PR adds support for linux and windows (gnu). I still need to investigate the reason why the msvc build fails but I currently don't have access to my main windows pc.
It also feature gates wxUSE_MEDIACTRL and wxUSE_WEBVIEW since these don't form part of the core gtk libs on linux, and actually require an installation of different libraries as well (gstream and libwebviewgtk respectively).
I've also added github actions for the 3 major platforms. 
Additionally, the build script now checks whether its running in a docs.rs server, and returns early in that case. That way the documentation should still be posted to docs.rs, without having to actually build wxwidgets.

AllenDang commented:
@MoAlyousef can you help to update README, add a short instruction about how to build and run on windows and linux?

MoAlyousef commented:
Hi @AllenDang 
Thank you. I've updated my fork and am testing some fixes for windows. I've also updated the README to add instructions on how to build for both platforms.
I'll create a PR once things are ready.

MoAlyousef commented:
The build fails on docs.rs since bindgen isn't called on the headers prior to returning from the build script. The current C wrapper headers require wxWidget headers so a simple fix would fail because it needs to download first which is not allowed  in docs.rs. 
I'm working on removing wxWidget headers from the C wrapper headers, and keeping them only in the source. That way, bindgen can be called on the C wrapper withouth having to download wxWidgets on docs.rs servers.
